python/lsst/pipe/base/graphBuilder.py

```diff
@@ -926,9 +926,12 @@ def makeGraph(self, pipeline, collections, run, userQuery,
         scaffolding = _PipelineScaffolding(pipeline, registry=self.registry)
         if not collections and (scaffolding.initInputs or scaffolding.inputs or scaffolding.prerequisites):
             raise ValueError("Pipeline requires input datasets but no input collections provided.")
-        instrument = pipeline.getInstrument()
-        if isinstance(instrument, str):
-            instrument = doImport(instrument)
+        instrument = None
+        if isinstance(pipeline, Pipeline):
+            instrument = pipeline.getInstrument()
+            if isinstance(instrument, str):
+                instrument = doImport(instrument)
+            pipeline = list(pipeline.toExpandedPipeline())
```

Comment:
Is it a problem that the `instrument` is always `None` if we pass an iterable of `TaskDef` instead of a `pipeline`?  Or does creating those taskdefs already do some magic with the instrument?